src/interactableComponents/drawer/index.js

```diff
@@ -0,0 +1,469 @@
+                  opacity: this.deltaX.interpolate({
+                    inputRange: [leftItemWidth - BLEED, leftItemWidth],
+                    outputRange: [0, 1],
+                    extrapolateLeft: 'clamp',
+                    extrapolateRight: 'clamp',
+                  }),
+                  transform: [{
+                    scale: this.deltaX.interpolate({
+                      inputRange: [leftItemWidth - BLEED, leftItemWidth],
+                      outputRange: [0.7, 1],
+                      extrapolateLeft: 'clamp',
+                      extrapolateRight: 'clamp',
+                    }),
+                  }],
+                },
+                ]}
+              >
+                {leftItem.text}
+              </Animated.Text>}
+            </View>
+          </TouchableHighlight>
+        </Animated.View>
+      </View>
+    );
+  }
+  renderRightItem(item, index) {
+    return (
+      <TouchableOpacity
+        key={index}
+        style={[
+          this.styles.button, {
+            width: this.getItemWidth(item),
+            minWidth: this.minItemWidth,
+            maxWidth: this.maxItemWidth,
+            backgroundColor: item.background || ITEM_BG,
+          },
+        ]}
+        onPress={item.onPress}
+        activeOpacity={item.onPress ? 0.7 : 1}
+      >
+        {item.icon &&
+        <Animated.Image
+          source={item.icon}
+          style={
+          [this.styles.buttonImage, {
+            opacity: this.deltaX.interpolate({
+              inputRange: this.inputRanges[index],
+              outputRange: [1, 0],
+              extrapolateLeft: 'clamp',
+              extrapolateRight: 'clamp',
+            }),
+            transform: [{
+              scale: this.deltaX.interpolate({
+                inputRange: this.inputRanges[index],
+                outputRange: [1, 0.7],
+                extrapolateLeft: 'clamp',
+                extrapolateRight: 'clamp',
+              }),
+            }],
+          },
+          ]}
+        />}
+        {item.text && 
+        <Animated.Text
+          numberOfLines={1}
+          style={
+          [this.styles.buttonText, {
+            opacity: this.deltaX.interpolate({
+              inputRange: this.inputRanges[index],
+              outputRange: [1, 0],
+              extrapolateLeft: 'clamp',
+              extrapolateRight: 'clamp',
+            }),
+            transform: [{
+              scale: this.deltaX.interpolate({
+                inputRange: this.inputRanges[index],
+                outputRange: [1, 0.7],
+                extrapolateLeft: 'clamp',
+                extrapolateRight: 'clamp',
+              }),
+            }],
+          },
+          ]}
+        >
+          {item.text}
+        </Animated.Text>}
+      </TouchableOpacity>
+    );
+  }
+  renderRightItems() {
+    const {height, rightItems} = this.props;
+
+    return (
+      <View style={{position: 'absolute', right: 0, height, flexDirection: 'row'}}>
```

Comment:
Let's avoid inline style wherever we can

Reply:
Done.